dist/tools/tunslip/tunslip6.c

```diff
@@ -147,18 +158,34 @@ stamptime(void)
 int
 is_sensible_string(const unsigned char *s, int len)
 {
-    int i;
+    if(len > 0) {
+        if (s[0] == 0x60) {
+            /* Possibly IPv6 packet with default traffic class, assume non-printable */
+            return 0;
+        }
+    }
 
-    for (i = 1; i < len; i++) {
+    int ret = 0;
+    for (int i = 1; i < len; i++) {
         if (s[i] == 0 || s[i] == '\r' || s[i] == '\n' || s[i] == '\t') {
             continue;
         }
         else if (s[i] < ' ' || '~' < s[i]) {
             return 0;
         }
+        /* only return 1 if the string contains at least one letter or number */
```

Comment:
Thanks for the clarification it definitely make sense :-)

Just an idea, what about just dumping all packets as an hexdump (see `hexdump -C`) so all of them are printed in the same way?

Example:
```
40 8b 88 48 5a cc 9a 35  8b 65 cb 96 75 f5 6e 74  |@..HZ..5.e..u.nt|
68 c1 82 05 3c f0 c0 03  5d bd 1b 22 22 22 22 49  |h...<...]..""""I|
```

(this relates as well to your comment: https://github.com/RIOT-OS/RIOT/pull/5677#discussion_r71969288)